src/bindings/python/flux/util.py

```diff
@@ -734,53 +806,115 @@ def filter_empty(self, items):
         empty value (as defined by the `empty` tuple defined below) for every
         entry in `items`.
         """
-        #  Build a list of all format strings that have the collapsible
-        #  sentinel '?:' to determine which are subject to the test for
-        #  emptiness.
+
+        #  Build a list of all format strings that have one of the width
+        #  adjustment sentinels to determine which are subject to the test for
+        #  emptiness/maxwidth.
+        #
+        sentinels = {"?:": "filter", "+:": "maxwidth", "?+:": "both"}
+
+        def sentinel_keys():
+            #  helper function to iterate supported sentinels by longest
+            #  first to avoid matching `+:` instead of `?+:`, etc.
+            #
+            return sorted(sentinels.keys(), key=lambda x: -len(x))
+
+        lst = []
+        index = 0
+
+        #  Loop over each format entry to find entries using any one
+        #  of the sentinels above:
         #
         #  Note: we remove the leading `text` and the format `spec` because
         #  these may make even empty formatted fields non-empty. E.g. a spec
         #  of `:>8` will always make the format result 8 characters wide.
         #
-        lst = []
-        index = 0
-        for text, field, _, conv in self.format_list:
-            if text.endswith("?:"):
+        for text, field, spec, conv in self.format_list:
+            # Determine if field has any supported sentinel above:
+            end = next((x for x in sentinel_keys() if text.endswith(x)), False)
+
+            if end:
+                #  This entry matches one of the filtering sentinels, parse
+                #  the spec to get current width (and allow possible
+                #  reconstruction later).
+                spec = self.FormatSpec(spec)
+
+                #  Save a format without any spec to allow deterimination of
+                #  maximum width after formatting all items:
```

Comment:
s/deterimination/determination/

Reply:
Thanks! Fixed that and I also dropped the flux-pgrep codeql commit since it didn't help workaround the codeql false positive. Will set MWP now.